File: examples/iOSDemo/Sources/DemoDesignerContext.swift
import RedlineShared
import SwiftUI

struct DemoDesignerContext: DesignerContext {
    func pins(screen: String, region: String) -> [DesignerPin] {
        switch region {
        case "Header":
            return [DesignerPin(component: "Title", pin: "size=lg")]
        case "Hero":
            return [DesignerPin(component: "Card", pin: "elevation=2")]
        case "CTA":
            return [DesignerPin(component: "Button", pin: "color=red, fillWidth=true")]
        default:
            return []
        }
    }
}

File: examples/AndroidDemo/src/main/java/dev/redline/android/demo/DemoDesignerContext.kt
package dev.redline.android.demo

import dev.redline.android.DesignerContext
import dev.redline.android.DesignerPin

object DemoDesignerContext : DesignerContext {
    override fun pins(screen: String, region: String): List<DesignerPin> = when (region) {
        "Header" -> listOf(DesignerPin(component = "Title", pin = "size=lg"))
        "Hero" -> listOf(DesignerPin(component = "Card", pin = "elevation=2"))
        "CTA" -> listOf(DesignerPin(component = "Button", pin = "color=red, fillWidth=true"))
        else -> emptyList()
    }
}
